src/components/ImportStatus.js

```diff
@@ -0,0 +1,73 @@
+import _ from 'lodash/fp'
+import { Fragment } from 'react'
+import { h } from 'react-hyperscript-helpers'
+import { clearNotification, notify } from 'src/components/Notifications'
+import { Ajax, useCancellation } from 'src/libs/ajax'
+import { pfbImportJobStore } from 'src/libs/state'
+import { delay, useAtom, useOnMount } from 'src/libs/utils'
+
+
+const ImportStatus = () => {
+  const jobs = useAtom(pfbImportJobStore)
+  return h(Fragment, _.map(job => h(ImportStatusItem, {
+    job,
+    onDone: () => {
+      pfbImportJobStore.update(_.reject({ jobId: job.jobId }))
+    }
+  }), jobs))
+}
+
+const ImportStatusItem = ({ job, onDone }) => {
+  const { targetWorkspace, jobId } = job
+  const signal = useCancellation()
+
+  useOnMount(() => {
+    const poll = async () => {
+      while (true) {
+        await delay(5000)
+        await isComplete(targetWorkspace, jobId)
+      }
+    }
+    poll()
+  })
+
+  const isComplete = async ({ namespace, name }, jobId) => {
+    const { Workspaces } = Ajax(signal)
+
+    try {
+      const response = await Workspaces.workspace(namespace, name).importPFBStatus(jobId)
+      const { message, status } = response
+
+      switch (status) {
```

Comment:
This is imperative code so Brett suggested `switch` instead of `Utils.switchCase()`. I understand why, but wow do I feel dirty having to remember to `break` at the end of every `case`!